src/NumericalAlgorithms/Interpolation/LinearLeastSquares.hpp

```diff
@@ -50,6 +50,9 @@ class LinearLeastSquares {
   template <typename T>
   std::array<double, Order + 1> fit_coefficients(const T& x_values,
                                                  const T& y_values);
+  template <typename T>
+  std::vector<std::array<double, Order + 1>> fit_coefficients(
```

Comment:
I think you need to document how the data is stored. That is, is each element of the `std::vector` holding a time series, eg `y_values[component][time]` or is it `y_values[time][component]`? I can't review the rest of the PR without knowing this since it changes how things are accessed! :)

You need to include `<vector>` now :)